src/core/namespace.c

```diff
@@ -531,11 +566,27 @@ static int mount_private_dev(MountEntry *m) {
                 goto fail;
         }
 
-        devptmx = strjoina(temporary_mount, "/dev/ptmx");
-        if (symlink("pts/ptmx", devptmx) < 0) {
+        /* /dev/ptmx can either be a device node or a symlink to /dev/pts/ptmx
+         * when /dev/ptmx a device node, /dev/pts/ptmx has 000 permissions making it inaccessible
+         * thus, in that case make a clone
+         *
+         * in nspawn and other containers it will be a symlink, in that case make it a symlink
+         */
+        if (lstat("/dev/ptmx", &st) == -1) {
                 r = -errno;
                 goto fail;
         }
+        if (S_ISLNK(st.st_mode)) {
```

Comment:
Does /dev/ptmx always exist? I prefer the following.
`if (lstat("/dev/ptmx", &st) < 0 || S_ISLNK(st.st_mode)) {`

Reply:
We assume `/dev/pts` does...  ptmx is always going to exist outside of a container.  It's from devtmpfs, and the Linux kernel ABI guarantee should apply.  Inside a container, systemd requires it to exist: https://www.freedesktop.org/wiki/Software/systemd/ContainerInterface/

Reply:
I thought that handles the failure of `lstat`?

Reply:
@sourcejdi OK. Thank you.